apps/system/js/statusbar.js

```diff
@@ -451,13 +451,82 @@ var StatusBar = {
   },
 
   _paused: 0,
-  pauseUpdate: function sb_pauseUpdate() {
-    this._paused++;
+
+  _pausedEvents: {
+    utilitytraywillshow: false,
+    utilitytrayopen: false,
+    utilitytraywillhide: false,
+    utilitytrayclose: false
   },
 
-  resumeUpdate: function sb_resumeUpdate() {
-    this._paused--;
-    this._updateIconVisibility();
+  pauseUpdate: function sb_pauseUpdate(evtType) {
+    if (this.shouldUpdate(evtType)) {
+      this._paused++;
+    }
+  },
+
+  resumeUpdate: function sb_resumeUpdate(evtType) {
+    if (this.shouldUpdate(evtType)) {
+      this._paused--;
+      this._updateIconVisibility();
```

Comment:
now that we will be confident in `this.isPaused()`, _updateIconVisibility should only get triggered here
`if (!this.isPaused())`